rmi/servercode/TransImpl/TransactionManager.java

```diff
@@ -36,11 +36,32 @@ public boolean abort(int id) throws InvalidTransactionException, TransactionAbor
 		if(toCommit.status == 1) {
 			toCommit.status = 0;
 			if(toCommit != null) {
```

Comment:
If we abort, we go one by one and reset our RM's by copying the latest saved shadow file. Just read the contents of "shadows/" + rm.banner + "_saved.ser", something like shadows/Cars_saved.ser